File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        if (nums == null || nums.length == 0) {
            return;
        }
        int low = 0;
        int mid = 0;
        int high = nums.length - 1;

        while (mid <= high) {
            if (nums[mid] == 1) {
                mid++;
            } else if (nums[mid] == 0) {
                swap(nums, mid, low);
                mid++;
                low++;
            } else {
                swap(nums, mid, high);
                high--;
            }
        }
    }

    private void swap(int[] nums, int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: 0075-sort-colors/0075-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """  
        low = 0
        mid = 0
        high = len(nums) -1

        while mid <= high:
            if nums[mid] == 1:
                mid = mid+1
            elif nums[mid] == 0:
                self.swap(nums,low,mid)
                low = low+1
                mid = mid+1
            else:
                self.swap(nums,mid,high)
                high = high-1
        
    def swap(self,nums,left,right):
        temp = nums[left]
        nums[left] = nums[right]
        nums[right] = temp